### PR TITLE
Refactor/collection list

### PR DESCRIPTION
**Pull request by theketan26**
## Description

Refactor: Collection list
Issue #738 

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build

## Have you tested locally?

- [x] 👍 yes
- [ ] 🙅 no, because I am lazy

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md

### CODE REVIEW

**Comment by punitkashyup** [comment]
Please attach the issue link

File: src/lib/components/collections/collections-list/import-curl/ImportCurl.viewModel.ts [hunk 1]
```diff
@@ -2,10 +2,4 @@ import { CollectionService } from "$lib/services/collection.service";
 
 export class ImportCurlViewModel {
   private collectionService = new CollectionService();
-
-  public importCollectionFromCurl = async (curl: string) => {
-    const response =
-      await this.collectionService.importCollectionFromCurl(curl);
-    return response;
-  };
 }
```
Comment (Astitva877) [inline src/lib/components/collections/collections-list/import-curl/ImportCurl.viewModel.ts:5]: If we have removed all methods from this file, and it doesn't serve any purpose so we can delete this file as well.

File: src/lib/components/collections/collections-list/searchTree/Request.svelte [hunk 2]
```diff
@@ -2,29 +2,26 @@
   import { replaceSlashWithGreaterThanSymbol } from "$lib/utils/helpers/common.helper";
   import { getMethodStyle } from "$lib/utils/helpers/conversion.helper";
   import { handleRequestClick } from "$lib/utils/helpers/handle-clicks.helper";
-  export let path: string = "";
+  export let explorer: any;
```
Comment (Astitva877) [inline src/lib/components/collections/collections-list/searchTree/Request.svelte:5]: We can make a interface to specify it's type, also we can change the name to something else which resonates to it's use.

**Review by Astitva877 — CHANGES_REQUESTED** [review]
@theketan26 comments are missing from various places, we can add comments for functions, imports to make the code more understandable.

File: src/pages/Collections/CollectionPage.ViewModel.ts [hunk 3]
```diff
@@ -330,4 +428,1594 @@ export class CollectionPageViewModel {
...
+    request: any,
+    folder: any,
+  ): Promise<boolean> => {
+    let userSource = {};
+    let workspaceId: string;
+    await this.getActiveWorkspace()
+      .subscribe((value) => (workspaceId = value._id))
+      .unsubscribe();
+    if (collection.activeSync) {
+      userSource = {
```
Comment (itsmdasifraza) [inline src/pages/Collections/CollectionPage.ViewModel.ts:1917]: pass workspaceId as  a parameter

File: src/pages/Collections/CollectionPage.ViewModel.ts [hunk 4]
```diff
@@ -330,4 +428,1594 @@ export class CollectionPageViewModel {
...
+    let collectionList = [];
+    let currentWorkspace;
+    await this.getCollectionList()
+      .subscribe((collections) => (collectionList = collections))
+      .unsubscribe();
+    await this.getActiveWorkspace()
+      .subscribe((workspace) => (currentWorkspace = workspace))
+      .unsubscribe();
+    const newCollection = {
+      id: UntrackedItems.UNTRACKED + uuidv4(),
```
Comment (itsmdasifraza) [inline src/pages/Collections/CollectionPage.ViewModel.ts:556]: get collection data from component as a props

File: src/pages/Collections/CollectionPage.ViewModel.ts [hunk 5]
```diff
@@ -330,4 +428,1594 @@ export class CollectionPageViewModel {
...
+
+  /**
+   * Handle input change in add collection popup
+   * @param importData: string - data to be validated
+   * @returns :{isimportDataLoading = boolean, isValidClientURL = boolean, isValidClientJSON = boolean, isValidClientXML = boolean, isValidServerURL = boolean, isValidServerJSON = boolean, isValidServerXML = boolean, isValidClientDeployedURL = boolean, isValidServerDeployedURL = boolean}
+   */
+  public handleInputDataChange = async (importData: string) => {
+    let isimportDataLoading = true;
+    let isValidClientURL = false;
+    let isValidClientJSON = false;
```
Comment (itsmdasifraza) [inline src/pages/Collections/CollectionPage.ViewModel.ts:1201]: isImportDataLoading should be handled in UI file

**Comment by punitkashyup** [comment]
Please attach the issue link

File: src/pages/Collections/CollectionPage.ViewModel.ts [hunk 6]
```diff
@@ -330,4 +429,1711 @@ export class CollectionPageViewModel {
       }
     }
   };
+
+  /**
+   * Handle updating tab
+   * @param data :any - tab data i.e. collection, folder or request
+   * @param route :string - path to collection, folder or request
+   * @param _id :string - if of the tab
+   */
```
Comment (itsmdasifraza) [inline src/pages/Collections/CollectionPage.ViewModel.ts:438]: type not required in js-doc comment as we are already using typescript, same at multiple places

File: src/pages/Collections/CollectionsPage.svelte [hunk 7]
```diff
@@ -124,57 +140,65 @@
...
+      onSearchCollection={_viewModel.handleSearchCollection}
+    />
   </Pane>
-  <Pane
-    class="sidebar-right-panel"
-    minSize={60}
-    size={$collapsibleState ? 100 : $collectionRightPanelWidth}
-  >
+  <Pane size={leftPanelCollapse ? 100 : rightPanelWidth} minSize={30}>
     <TabBar
```
Comment (itsmdasifraza) [inline src/pages/Collections/CollectionsPage.svelte:194]: set minSize as earlier

File: src/pages/Collections/CollectionsPage.svelte [hunk 8]
```diff
@@ -124,57 +140,65 @@
...
 
 <Splitpanes
   class="splitter-sidebar"
   on:resize={(e) => {
-    collectionLeftPanelWidth.set(e.detail[0].size);
-    collectionRightPanelWidth.set(e.detail[1].size);
+    leftPanelWidth = e.detail[0].size;
+    rightPanelWidth = e.detail[1].size;
   }}
 >
```
Comment (itsmdasifraza) [inline src/pages/Collections/CollectionsPage.svelte:169]: local variable is not persistent here

**Review by Astitva877 — APPROVED** [review]

**Pull request merged**